dm/master/server.go

```diff
@@ -1332,6 +1349,62 @@ func (s *Server) generateSubTask(ctx context.Context, task string) (*config.Task
 	return cfg, stCfgs, nil
 }
 
+func (s *Server) removeMetaData(ctx context.Context, cfg *config.TaskConfig) error {
+	toDB := *cfg.TargetDB
+	toDB.Adjust()
+	if len(toDB.Password) > 0 {
+		pswdTo, err := utils.Decrypt(toDB.Password)
+		if err != nil {
+			return err
+		}
+		toDB.Password = pswdTo
+	}
+
+	sqls := make([]string, 0, 3)
+	sqls = append(sqls, fmt.Sprintf("CREATE SCHEMA IF NOT EXISTS `%s`", cfg.MetaSchema))
+	// clear loader and syncer checkpoints
+	sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`%s`",
+		cfg.MetaSchema, cfg.Name+"_loader_checkpoint"))
+	sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`%s`",
+		cfg.MetaSchema, cfg.Name+"_syncer_checkpoint"))
+
+	// clear etcd data
+	if cfg.ShardMode == config.ShardPessimistic {
+		err := s.pessimist.RemoveMetaData(cfg.Name)
+		if err != nil {
+			return err
+		}
+		// clear shard meta data for pessimistic
+		sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`%s`",
+			cfg.MetaSchema, fmt.Sprintf(shardmeta.MetaTableFormat, cfg.Name)))
+	} else if cfg.ShardMode == config.ShardOptimistic {
+		err := s.optimist.RemoveMetaData(cfg.Name)
+		if err != nil {
+			return err
+		}
+	}
+
+	// clear online ddl schema
+	if cfg.OnlineDDLScheme != "" {
+		sqls = append(sqls, fmt.Sprintf("DROP TABLE IF EXISTS `%s`.`%s`",
+			cfg.MetaSchema, fmt.Sprintf("%s_onlineddl", cfg.Name)))
+	}
+
+	baseDB, err := conn.DefaultDBProvider.Apply(toDB)
+	if err != nil {
+		return terror.WithScope(err, terror.ScopeDownstream)
+	}
+	defer baseDB.Close()
+	dbConn, err := baseDB.GetBaseConn(ctx)
+	if err != nil {
+		return terror.WithScope(err, terror.ScopeDownstream)
+	}
+	defer baseDB.CloseBaseConn(dbConn)
+	ctctx := tcontext.Background().WithContext(ctx).WithLogger(log.With(zap.String("unit", "remove metadata")))
+	_, err = dbConn.ExecuteSQL(ctctx, stmtHistogram, cfg.Name, []string{})
```

Comment:
do you mean execute `sqls`?

Reply:
and these sqls used to drop table, do we need to collect the metric?

Reply:
Add a judgement to test whether `hVec` is `nil` in `baseconn.go`?

Reply:
I think it's ok